plugins/randomforest_procedure.cc

```diff
@@ -0,0 +1,339 @@
+#include "mldb/vfs/fs_utils.h"
+
+#include <boost/random/mersenne_twister.hpp>
+#include <boost/random/uniform_int.hpp>
+#include <boost/random/uniform_01.hpp>
+
+using namespace std;
+using namespace ML;
+
+namespace Datacratic {
+namespace MLDB {
+
+DEFINE_STRUCTURE_DESCRIPTION(RandomForestProcedureConfig);
+
+RandomForestProcedureConfigDescription::
+RandomForestProcedureConfigDescription()
+{
+    addField("trainingData", &RandomForestProcedureConfig::trainingData,
+             "Specification of the data for input to the classifier procedure. "
+             "The select expression must contain these two sub-expressions: one row expression "
+             "to identify the features on which to train and one scalar expression "
+             "to identify the label.  The type of the label expression must match "
+             "that of the classifier mode: a boolean (0 or 1) for `boolean` mode; "
+             "a real for regression mode, and any combination of numbers and strings "
+             "for `categorical` mode.  Labels with a null value will have their row skipped. "
+             "The select statement does not support groupby and having clauses. "
+             "Also, unlike most select expressions, this one can only select whole columns, "
+             "not expressions involving columns. So X will work, but not X + 1. "
+             "If you need derived values in the select expression, create a dataset with "
+             "the derived columns as a previous step and run the classifier over that dataset instead.");
+    addField("modelFileUrl", &RandomForestProcedureConfig::modelFileUrl,
+             "URL where the model file (with extension '.cls') should be saved. "
+             "This file can be loaded by a function of type 'classifier'.");   
+    addField("featureVectorSamplings", &RandomForestProcedureConfig::featureVectorSamplings,
+             "Number of samplings of feature vectors. "
+             "The total number of bags will be featureVectorSamplings*featureSamplings.", 5);
+    addField("featureVectorSamplingProp", &RandomForestProcedureConfig::featureVectorSamplingProp,
+             "Proportion of feature vectors to select in each sample. ", 0.3f);
+    addField("featureSamplings", &RandomForestProcedureConfig::featureSamplings,
+             "Number of samplings of features. "
+             "The total number of bags will be featureVectorSamplings*featureSamplings.", 20);
+    addField("featureSamplingProp", &RandomForestProcedureConfig::featureSamplingProp,
+             "Proportion of features to select in each sample. ", 0.3f);
+    addField("maxDepth", &RandomForestProcedureConfig::maxDepth,
+             "Maximum depth of the trees ", 20);   
+    addField("functionName", &RandomForestProcedureConfig::functionName,
+             "If specified, a classifier function of this name will be created using "
+             "the trained classifier.");
+    addField("verbosity", &RandomForestProcedureConfig::verbosity,
```

Comment:
The documentation on that field is not sufficient.  If we plan to support a fixed range of integer than we should be explicit about it.  For the time being it seems that a boolean would be better.